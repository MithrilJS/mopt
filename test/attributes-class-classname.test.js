"use strict";

var code = require("./lib/code"),

    selectors = {
        tag   : `"selector"`,
        class : `".selector"`,
        tmpl  : `\`template\``,

        "class + attr" : `".selector[attr]"`
    },

    classes = {
        missing : false,
        empty   : `""`,
        value   : `"class"`,
        tmpl    : `\`class\``
    },

    classnames = {
        missing : false,
        empty   : `""`,
        value   : `"className"`,
        tmpl    : `\`className\``
    },
    
    strings = {
        missing : false,
        value   : `"inner"`
    };

describe("Attributes", () => {
    describe("class vs className", () => {
        var combinations = [];

        Object.keys(selectors).forEach((selectorval) =>
            Object.keys(classes).forEach((classval) =>
                Object.keys(classnames).forEach((classnameval) =>
                    Object.keys(strings).forEach((stringval) => {
                        combinations.push({
                            name      : `selector ${selectorval}, class ${classval}, className ${classnameval}, string ${stringval}`,
                            selector  : selectors[selectorval],
                            class     : classes[classval],
                            classname : classnames[classnameval],
                            string    : strings[stringval]
                        });
                    })
                )
            )
        );

        combinations.forEach((test) => {
            it(test.name, () => {
                var str = "",
                    
                    hasClass     = typeof test.class === "string",
                    hasClassName = typeof test.classname === "string",
                    hasString    = typeof test.string === "string";

                str += `m(${test.selector}`;

                if(hasClass || hasClassName) {
                    str += `, {`;

                    if(hasClass) {
                        str += `class: ${test.class}`;
                    }

                    if(hasClassName) {
                        if(hasClass) {
                            str += `, `;
                        }

                        str += `className: ${test.classname}`;
                    }

                    str += `}`;
                }

                if(hasString) {
                    str += `, ${test.string}`;
                }

                str += `);`;

                expect(code(str)).toMatchSnapshot();
            });
        });
    });
});
