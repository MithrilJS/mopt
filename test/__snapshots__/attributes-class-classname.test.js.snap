// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attributes class vs className selector class + attr, class empty, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class empty, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class empty, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class empty, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class empty, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`,attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class empty, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`,attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class empty, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class empty, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`,attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`,attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class missing, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`,attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`,attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`,attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`,attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+(\\" \\"+\`className\`),attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+(\\" \\"+\`className\`),attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+\\" className\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class tmpl, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+\\" className\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class \\"+\`className\`,attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class \\"+\`className\`,attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class className\\",attribute:true},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class + attr, class value, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class className\\",attribute:true},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class empty, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class empty, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class empty, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class empty, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class empty, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class empty, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class empty, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class empty, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class missing, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class missing, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class missing, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class missing, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class missing, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class missing, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`className\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class missing, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class missing, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+(\\" \\"+\`className\`)},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+(\\" \\"+\`className\`)},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+\\" className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class tmpl, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector \\"+\`class\`+\\" className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class value, className empty, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class value, className empty, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class value, className missing, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class value, className missing, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class value, className tmpl, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class \\"+\`className\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class value, className tmpl, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class \\"+\`className\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector class, class value, className value, string missing 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector class, class value, className value, string value 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"selector class className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className empty, string missing 1`] = `"m.vnode(\\"selector\\",undefined,undefined,[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className empty, string value 1`] = `"m.vnode(\\"selector\\",undefined,undefined,undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className missing, string missing 1`] = `"m.vnode(\\"selector\\",undefined,undefined,[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className missing, string value 1`] = `"m.vnode(\\"selector\\",undefined,undefined,undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className tmpl, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`className\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className tmpl, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`className\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className value, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class empty, className value, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className empty, string missing 1`] = `"m.vnode(\\"selector\\",undefined,undefined,[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className empty, string value 1`] = `"m.vnode(\\"selector\\",undefined,undefined,undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className missing, string missing 1`] = `"m.vnode(\\"selector\\",undefined,undefined,[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className missing, string value 1`] = `"m.vnode(\\"selector\\",undefined,undefined,undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className tmpl, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`className\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className tmpl, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`className\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className value, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class missing, className value, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className empty, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className empty, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className missing, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className missing, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className tmpl, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`+(\\" \\"+\`className\`)},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className tmpl, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`+(\\" \\"+\`className\`)},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className value, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`+\\" className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class tmpl, className value, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\`class\`+\\" className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class value, className empty, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class value, className empty, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class value, className missing, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class value, className missing, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class value, className tmpl, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class \\"+\`className\`},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class value, className tmpl, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class \\"+\`className\`},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tag, class value, className value, string missing 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class className\\"},[],undefined,undefined);"`;

exports[`Attributes class vs className selector tag, class value, className value, string value 1`] = `"m.vnode(\\"selector\\",undefined,{className:\\"class className\\"},undefined,\\"inner\\",undefined);"`;

exports[`Attributes class vs className selector tmpl, class empty, className empty, string missing 1`] = `"m(\`template\`,{class:\\"\\",className:\\"\\"});"`;

exports[`Attributes class vs className selector tmpl, class empty, className empty, string value 1`] = `"m(\`template\`,{class:\\"\\",className:\\"\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class empty, className missing, string missing 1`] = `"m(\`template\`,{class:\\"\\"});"`;

exports[`Attributes class vs className selector tmpl, class empty, className missing, string value 1`] = `"m(\`template\`,{class:\\"\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class empty, className tmpl, string missing 1`] = `"m(\`template\`,{class:\\"\\",className:\`className\`});"`;

exports[`Attributes class vs className selector tmpl, class empty, className tmpl, string value 1`] = `"m(\`template\`,{class:\\"\\",className:\`className\`},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class empty, className value, string missing 1`] = `"m(\`template\`,{class:\\"\\",className:\\"className\\"});"`;

exports[`Attributes class vs className selector tmpl, class empty, className value, string value 1`] = `"m(\`template\`,{class:\\"\\",className:\\"className\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class missing, className empty, string missing 1`] = `"m(\`template\`,{className:\\"\\"});"`;

exports[`Attributes class vs className selector tmpl, class missing, className empty, string value 1`] = `"m(\`template\`,{className:\\"\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class missing, className missing, string missing 1`] = `"m(\`template\`);"`;

exports[`Attributes class vs className selector tmpl, class missing, className missing, string value 1`] = `"m(\`template\`,\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class missing, className tmpl, string missing 1`] = `"m(\`template\`,{className:\`className\`});"`;

exports[`Attributes class vs className selector tmpl, class missing, className tmpl, string value 1`] = `"m(\`template\`,{className:\`className\`},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class missing, className value, string missing 1`] = `"m(\`template\`,{className:\\"className\\"});"`;

exports[`Attributes class vs className selector tmpl, class missing, className value, string value 1`] = `"m(\`template\`,{className:\\"className\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className empty, string missing 1`] = `"m(\`template\`,{class:\`class\`,className:\\"\\"});"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className empty, string value 1`] = `"m(\`template\`,{class:\`class\`,className:\\"\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className missing, string missing 1`] = `"m(\`template\`,{class:\`class\`});"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className missing, string value 1`] = `"m(\`template\`,{class:\`class\`},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className tmpl, string missing 1`] = `"m(\`template\`,{class:\`class\`,className:\`className\`});"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className tmpl, string value 1`] = `"m(\`template\`,{class:\`class\`,className:\`className\`},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className value, string missing 1`] = `"m(\`template\`,{class:\`class\`,className:\\"className\\"});"`;

exports[`Attributes class vs className selector tmpl, class tmpl, className value, string value 1`] = `"m(\`template\`,{class:\`class\`,className:\\"className\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class value, className empty, string missing 1`] = `"m(\`template\`,{class:\\"class\\",className:\\"\\"});"`;

exports[`Attributes class vs className selector tmpl, class value, className empty, string value 1`] = `"m(\`template\`,{class:\\"class\\",className:\\"\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class value, className missing, string missing 1`] = `"m(\`template\`,{class:\\"class\\"});"`;

exports[`Attributes class vs className selector tmpl, class value, className missing, string value 1`] = `"m(\`template\`,{class:\\"class\\"},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class value, className tmpl, string missing 1`] = `"m(\`template\`,{class:\\"class\\",className:\`className\`});"`;

exports[`Attributes class vs className selector tmpl, class value, className tmpl, string value 1`] = `"m(\`template\`,{class:\\"class\\",className:\`className\`},\\"inner\\");"`;

exports[`Attributes class vs className selector tmpl, class value, className value, string missing 1`] = `"m(\`template\`,{class:\\"class\\",className:\\"className\\"});"`;

exports[`Attributes class vs className selector tmpl, class value, className value, string value 1`] = `"m(\`template\`,{class:\\"class\\",className:\\"className\\"},\\"inner\\");"`;
